pkg/istio-agent/xds_proxy.go

```diff
@@ -444,6 +449,18 @@ func (p *XdsProxy) handleUpstreamRequest(con *ProxyConnection) {
 				return
 			}
 
+			// override the first xds request node metadata labels
```

Comment:
Overall I am pretty concerned about mutating the incoming XDS requests... a few possible concerns:

1. istiod 1.16 will need to support proxy 1.15, which do not have this code, so the new label stuff won't work which may lead to incorrect config (including authz)
2. Possible performance impact of mutating request each time. Not likely a major concern though.
3. Only works in xds proxy; this means other clients (proxyless grpc, etc) will be broken still which can then lead to RBAC bypass.

Reply:
I think 1 and 2 are not big problem, since this is a new feature.
For 3, I think grpc should so something in itself to reload metadata?

Reply:
1 is a problem. Labels are an existing feature, and we are changing how they work. if we now incorrectly determine the set of labels, that could be a CVE. Same with 2 -- this impacts everyone, not just users mutating labels.

on 3 - If we are changing requirements on clients maybe we should have wider discussion outside this PR?

Reply:
As you said labels update by design has no problem(k8s long support labels update), incorrectly determine labels could cause some bug or CVE. 

For 2, this is why I only regen labels only instead if all meta. Not a big cost though, it is run in proxy rather than istiod.

For 3, we should have a document about labels behavior.